src/libYARP_OS/src/Contact.cpp

```diff
@@ -197,6 +198,87 @@ Contact Contact::fromString(const ConstString& txt)
     return c;
 }
 
+bool Contact::isValidRegistrationName(const char* name)
+{
+    if(strcmp(name,"") == 0)                                    // anonymous
+        return true;
+
+    const std::string port_name_s(name);
+    std::regex port_exp("([a-zA-Z0-9]+(?:[_][a-zA-Z0-9]+)*)"    // plain expression "abc[_def]" without leading '/', allowed for topics
+                        "|"                                     // OR
+                        "(\\.\\.\\.)");                         // "..." is accepted as registration name, a default one is assigned
+
+    std::smatch port_match;
+    if (std::regex_match(port_name_s, port_match, port_exp))
+        return true;
+
+    return isValidRegisteredName(name);                         // otherwise, we need a good final name
+}
+
+bool Contact::isValidRegisteredName(const char* name)
+{
+    const std::string port_name_s(name);
+    std::regex port_exp(
```

Comment:
Do you have any idea about the order of magnitute necessary to build the `std::regex` object? If it is not trivial, we could think about having a regex singleton to avoid rebuilding the regex for each check.